pkgs/tools/admin/manta/default.nix

```diff
@@ -0,0 +1,62 @@
+{ lib, buildNpmPackage, fetchurl, nodejs, installShellFiles, testers, node-manta }:
+
+let
+  source = builtins.fromJSON (builtins.readFile ./source.json);
+in
+buildNpmPackage rec {
+  pname = "manta";
+  inherit (source) version;
+
+  src = fetchurl {
+    url = "https://registry.npmjs.org/${pname}/-/${source.filename}";
+    hash = source.integrity;
+  };
+
+  npmDepsHash = source.deps;
+
+  dontNpmBuild = true;
+
+  nativeBuildInputs = [ nodejs installShellFiles ];
+
+  postPatch = ''
+    # Use generated package-lock.json as upstream does not provide one
+    ln -s ${./package-lock.json} package-lock.json
+  '';
+
+  postInstall = ''
+    mkdir -p $out/bin
+
+    cmds=$(find $out/lib/node_modules/manta/bin/ -type f -printf "%f\n")
+
+    for cmd in $cmds; do
+      ln -s ../lib/node_modules/manta/bin/$cmd $out/bin/$cmd
+    done
+  '';
+
+  postFixup = ''
+    # create completions, following upstream procedure https://github.com/joyent/node-manta/blob/v5.3.2/Makefile#L85-L91
+    cmds=$(find ./bin/ -type f -printf "%f\n")
+
+    node $out/lib/node_modules/manta/lib/create_client.js
+
+    for cmd in $cmds; do
+      installShellCompletion --cmd $cmd --bash <($out/bin/$cmd --completion)
+    done
+  '';
+
+  passthru = {
+    tests.version = testers.testVersion {
+      package = node-manta;
+    };
+    updateScript = ./update.sh;
+  };
+
+  meta = with lib; {
+    description = "Manta Client CLIs and Node.js SDK (https://www.tritondatacenter.com/triton/object-storage)";
+    homepage = "https://github.com/TritonDataCenter/node-manta";
+    license = licenses.mit;
+    platforms = platforms.all;
```

Comment:
`buildNpmPackage` sets this by default, you should only set it if you have a good reason to do so. (Plus, `all` wouldn't even be correct here, I don't think.)

Reply:
```suggestion
```

Reply:
Fixed, thanks.